lib/index.ts

```diff
@@ -0,0 +1,50 @@
+import { clientInfo } from './environment'
+
+declare var PercyAgent: any;
+
+/**
+ * A function to take a Percy snapshot from a Puppeteer test or script. To use in your tests:
+ *   const { percySnapshot } = require('@percy/puppeteer')
+ *
+ *   const browser = await puppeteer.launch()
+ *   const page = await browser.newPage()
+ *   await page.goto(<your.test.url>)
+ *   await percySnapshot(page, <your snapshot name>, <maybe options>)
+ *
+ * @param page Puppeteer Page object that we are snapshotting. Required.
+ * @param name Name of the snapshot that we're taking. Required.
+ * @param options Additional options, e.g. '{widths: [768, 992, 1200]}'. Optional.
+ */
+export async function percySnapshot(page: any, name: string, options: any = {}) {
+  if (!page) {
```

Comment:
I'm not 100% sure how idiomatic is this style of checking for required parameters in JavaScript. Let me know if this seems overkill or if you see a better way to do this.

Reply:
I think that's fine

Reply:
In TypeScript land this is already a required param so no worries there.

I'm not sure what the resulting JS looks like and what sort of error you would get if we didn't have this check in place. My guess is you might get something like "addScriptTag is not a function of undefined" which is not super obvious. The error message you have is pretty nice and friendly :-)

Reply:
Yeah, that's exactly what happens if I don't `throw` explicitly here -- you would get a more obscure error message later in the execution.